examples/cpp/stereo_vision_slam/README.md

```diff
@@ -0,0 +1,182 @@
+        return Collection<rerun::Position3D>::take_ownership(std::move(positions));
+    }
+};
+
+
+template <>
+struct rerun::CollectionAdapter<rerun::Position3D, Eigen::Matrix3Xf>
+{
+    /* Adapters so we can log an eigen matrix as rerun positions */
+
+    // Sanity check that this is binary compatible.
+    static_assert(
+        sizeof(rerun::Position3D) == sizeof(Eigen::Matrix3Xf::Scalar) * Eigen::Matrix3Xf::RowsAtCompileTime
+    );
+
+    Collection<rerun::Position3D> operator()(const Eigen::Matrix3Xf& matrix)
+    {
+        // Borrow for non-temporary.
+        static_assert(alignof(rerun::Position3D) <= alignof(Eigen::Matrix3Xf::Scalar));
+        return Collection<rerun::Position3D>::borrow(
+            // Cast to void because otherwise Rerun will try to do above sanity checks with the wrong type (scalar).
+            reinterpret_cast<const void*>(matrix.data()),
+            matrix.cols()
+        );
+    }
+
+    Collection<rerun::Position3D> operator()(Eigen::Matrix3Xf&& matrix)
+    {
+        /* Do a full copy for temporaries (otherwise the
+         * data might be deleted when the temporary is destroyed). */
+        std::vector<rerun::Position3D> positions(matrix.cols());
+        memcpy(positions.data(), matrix.data(), matrix.size() * sizeof(rerun::Position3D));
+        return Collection<rerun::Position3D>::take_ownership(std::move(positions));
+    }
+};
+
+```
+
+## Images
+```cpp
+// Draw stereo left image
+rec.log(entity_name,
+        rerun::Image(tensor_shape(kf_sort[0].second->left_img_),
+                    rerun::TensorBuffer::u8(kf_sort[0].second->left_img_)));
+```
+
+## Pinhole camera
+
+The camera frames shown in the space view is generated by the following code:
+
+```cpp
+rec.log(entity_name,
+    rerun::Transform3D(
+        rerun::Vec3D(camera_position.data()),
+        rerun::Mat3x3(camera_orientation.data()), true)
+);
+// …
+rec.log(entity_name,
+        rerun::Pinhole::from_focal_length_and_resolution({fx, fy}, {img_num_cols, img_num_rows}));
+```
+
+## Time series
+```cpp
+void Viewer::Plot(std::string plot_name, double value, unsigned long maxkeyframe_id)
+{
+    // …
+    rec.set_time_sequence("max_keyframe_id", maxkeyframe_id);
+    rec.log(plot_name, rerun::Scalar(value));
+}
+```
+
+## Trajectory
+```cpp
+rec.log("world/path",
+    rerun::Transform3D(
+        rerun::Vec3D(camera_position.data()),
+        rerun::Mat3x3(camera_orientation.data()), true));
+
+std::vector<rerun::datatypes::Vec3D> path;
+// …
+rec.log("world/path", rerun::LineStrips3D(rerun::LineStrip3D(path)));
+```
+
+## Point cloud
+```cpp
+rec.log("world/landmarks",
+    rerun::Transform3D(
+        rerun::Vec3D(camera_position.data()),
+        rerun::Mat3x3(camera_orientation.data()), true));
+
+std::vector<Eigen::Vector3f> points3d_vector;
+// …
+rec.log("world/landmarks", rerun::Points3D(points3d_vector));
+```
+
+## Text log
+
+```cpp
+rec.log("world/log", rerun::TextLog(msg).with_color(log_color.at(log_type)));
```

Comment:
Wha is this weird `log_color.at(log_type)` thing?

Reply:
`log_color` is a dictionary that maps the type of text log to a color. For example logs related to loop-closure will have a different color than those related to the frontend.